### PR TITLE
Multi-language Capability

### PR DESCRIPTION
**Pull request by morgendigital**
We propose an integration of an internationalization library (i18n) to allow the UI of Chaindesk to be used in multiple languages. Translation can easily be done over a JSON file, and a language switcher is included in the settings.

Is there any interest for such a functionality?

### CODE REVIEW

**Comment by morgendigital** [comment]
That sounds great @gmpetrov - thank you. We will prepare a test setup over the next couple of days and will get back to you.

**Comment by morgendigital** [comment]
@gmpetrov have you checked our recent implementation that we would have liked to commit in the first place? Is there any feedback you can give us here already? Then we can improve it in the test implementation you proposed. Thank you :)

**Comment by gmpetrov** [comment]
> @gmpetrov have you checked our recent implementation that we would have liked to commit in the first place? Is there any feedback you can give us here already? Then we can improve it in the test implementation you proposed. Thank you :)

Hello @morgendigital we're going to check ASAP 

@OdapX can you review this PR pls?

**Comment by morgendigital** [comment]
Thanks, both of you. Just let us know if there's something you'd like to adapt, we will quickly implement it and draft a new round.

File: Dockerfile [hunk 1]
```diff
@@ -36,7 +36,7 @@ COPY turbo.json turbo.json
 # Uncomment the following line in case you want to disable telemetry during the build.
 ENV NEXT_TELEMETRY_DISABLED 1
 # RUN mv next.config.docker.js next.config.js
-# RUN npm run prisma:generate
+# RUN npm run prisma:generateBzw.
```
Comment (OdapX) [inline Dockerfile:39]: to drop

File: apps/dashboard/components/AgentDeployTab.tsx [hunk 2]
```diff
@@ -103,8 +105,8 @@ function AgentDeployTab(props: Props) {
   return (
     <>
       <SettingCard
-        title="Deploy"
-        description="Deploy your agent with the following widgets or integrations"
+        title={t('publish')}
+        description={t('publish-subtitle')}
```
Comment (OdapX) [inline apps/dashboard/components/AgentDeployTab.tsx:109]: use camelcase, applies elsewhere too.

File: apps/dashboard/components/AgentInputs/ModelInput.tsx [hunk 3]
```diff
@@ -229,48 +225,42 @@ export default function ModelInput({}: Props) {
       {/* <Divider /> */}
 
       <Stack sx={{ py: 2 }} gap={1}>
-        <Typography level="title-md">Behavior</Typography>
+        <Typography level="title-md">{t('behavior')}</Typography>
 
         <FormControl>
-          <FormLabel>Knowledge Restriction</FormLabel>
+          <FormLabel>{t('knowRest')}</FormLabel>
```
Comment (OdapX) [inline apps/dashboard/components/AgentInputs/ModelInput.tsx:231]: please be as explicit as possible with the keys. (knowledgeRestriction)

File: apps/dashboard/components/DatastoreForms/Base.tsx [hunk 4]
```diff
@@ -110,7 +112,7 @@ export default function BaseForm(props: Props) {
         {networkError && <Alert color="danger">{networkError}</Alert>}
 
         <Input
-          label="Datastore Name"
+          label={t('nameKonfig')}
```
Comment (OdapX) [inline apps/dashboard/components/DatastoreForms/Base.tsx:115]: datastoreName

File: apps/dashboard/components/EmailInboxSettingsTab.tsx [hunk 5]
```diff
@@ -75,13 +77,13 @@ function EmailInboxSettingsTab({ inboxId }: Props) {
                   }
                 },
                 color: 'danger',
-                children: 'Delete',
+                children: `${t('deleteCta')}`,
```
Comment (OdapX) [inline apps/dashboard/components/EmailInboxSettingsTab.tsx:80]: should not be structured this way. ctas should be a nested object where relevant.

File: apps/dashboard/components/Layout/Navigation.tsx [hunk 6]
```diff
@@ -370,15 +372,15 @@ export default function Navigation() {
       //   active: router.route === RouteNames.APPS,
       // },
       {
-        label: 'Settings',
-        route: RouteNames.SETTINGS,
+        label: t('settings'),
+        route: RouteNames.PROFILE,
         icon: <ManageAccountsRoundedIcon fontSize="small" />,
-        active: router.route.startsWith(RouteNames.SETTINGS),
+        active: router.route.startsWith(RouteNames.PROFILE),
```
Comment (OdapX) [inline apps/dashboard/components/Layout/Navigation.tsx:378]: why this change?

File: apps/dashboard/components/Layout/index.tsx [hunk 7]
```diff
@@ -52,7 +54,13 @@ export default function Layout(props: Props) {
 
   React.useEffect(() => {
     setMounted(true);
-  }, []);
+    if (session?.user?.language) {
```
Comment (OdapX) [inline apps/dashboard/components/Layout/index.tsx:57]: I think, i18next has a plugin for automatic locale detection. use i18next-browser-languagedetector

**Comment by JanikHalder** [comment]
Hi,

We made this change to solve a Routing Issue which leads to a re-rendering of the whole UI instead of only the necassary parts. 

It was linked to https://app.chaindesk.ai/settings/, but this URL only redirects to https://app.chaindesk.ai/settings/profile. By replacing the URL with /settings/profile it is possible to prevent the whole page from reloading. 

Best Janik.

**Comment by OdapX** [comment]
Hi, thank you for your contribution!

I have the following remarks:
   - use camel case for all keys (no mixing), and carefull with spelling (eg, nameKonfig)
   - be very explicit with the key names (even if the key is long).
   - scope down the PR to just translation changes:  do not add language to user schema (lang would probably be added to the org later),do not perform updates and do not rename settings to profile.
   - Structure: Not sure if this is the best structure for the long run, we still need to think about a clean way to do this.
   
   I think as @gmpetrov suggested, you could scope this to just one page and no need to change the schema at the moment (use this plugin i18next-browser-languagedetector to detect the locale).

  thanks.

**Comment by morgendigital** [comment]
Thank you @OdapX for your extensive and detailed comment! We are on it!
